chain/x/zoracle/internal/keeper/querier.go

```diff
@@ -230,3 +234,72 @@ func queryPending(ctx sdk.Context, path []string, req abci.RequestQuery, keeper
 func queryRequestNumber(ctx sdk.Context, req abci.RequestQuery, keeper Keeper) ([]byte, sdk.Error) {
 	return codec.MustMarshalJSONIndent(keeper.cdc, keeper.GetRequestCount(ctx)), nil
 }
+
+func queryDataSource(ctx sdk.Context, path []string, req abci.RequestQuery, keeper Keeper) ([]byte, sdk.Error) {
+	if len(path) == 0 {
+		return nil, sdk.ErrInternal("must specify the data source id")
+	}
+	id, err := strconv.ParseInt(path[0], 10, 64)
+	if err != nil {
+		return nil, sdk.ErrInternal(fmt.Sprintf("wrong format for data source id %s", err.Error()))
+	}
+
+	dataSource, sdkErr := keeper.GetDataSource(ctx, id)
+	if sdkErr != nil {
+		return nil, sdkErr
+	}
+
+	res, err := codec.MarshalJSONIndent(keeper.cdc, types.NewDataSourceQuerierInfo(
+		id,
+		dataSource.Owner,
+		dataSource.Name,
+		dataSource.Fee,
+		dataSource.Executable,
+	))
+	if err != nil {
+		panic("could not marshal result to JSON")
+	}
+
+	return res, nil
+}
+
+func queryDataSources(ctx sdk.Context, path []string, req abci.RequestQuery, keeper Keeper) ([]byte, sdk.Error) {
+	if len(path) != 2 {
+		return nil, sdk.ErrInternal("must specify the data source start_id and number of data sources")
+	}
+	startID, err := strconv.ParseInt(path[0], 10, 64)
+	if err != nil {
+		return nil, sdk.ErrInternal(fmt.Sprintf("wrong format for data source start id %s", err.Error()))
+	}
+
+	numberOfDataSources, err := strconv.ParseInt(path[1], 10, 64)
+	if err != nil {
+		return nil, sdk.ErrInternal(fmt.Sprintf("wrong format for number of data sources %s", err.Error()))
+	}
+	if numberOfDataSources < 1 || numberOfDataSources > 100 {
+		return nil, sdk.ErrInternal("number of data sources should be >= 1 and <= 100")
+	}
+
+	dataSources := []types.DataSourceQuerierInfo{}
+	for offset := int64(0); offset < numberOfDataSources; offset++ {
+		id := startID + offset
+		dataSource, sdkErr := keeper.GetDataSource(ctx, id)
+		if sdkErr != nil {
+			break
```

Comment:
You see this is bad. You should not just throw away errors without checking that it is exactly what you expect.